org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/CompletionEngine.java

```diff
@@ -2331,7 +2331,13 @@ public void complete(ICompilationUnit sourceUnit, int completionPosition, int po
 
 						if ((this.unitScope = parsedUnit.scope) != null) {
 							this.lookupEnvironment.completeTypeBindings(parsedUnit, true);
+							if (this.unitScope.deferredException != null) {
+								throw this.unitScope.deferredException;
+							}
```

Comment:
This exception is not cleared anywhere - is that ok? Are all the scopes involved transactional and transient to not matter ?

Reply:
The only client interested in `unitScope.deferredException` is this very code block in `CompletionEngine`. Once the exception has been picked up an thrown, nobody will ever look at it again. But I agree it would look nicer, if throwing would also consume the exception - I'll change that.